  @Test(dataProvider = "invalidEnvironmentProvider")
  public void shouldFailValidationIfEnvironmentIsBroken(
      EnvironmentDto env, String expectedExceptionMessage) throws Exception {

    try {
      // when
      environmentValidator.validate("env", env);

      // then
      fail(format("Validation had to throw exception with message %s", expectedExceptionMessage));
    } catch (IllegalArgumentException e) {
      // then
      assertEquals(e.getLocalizedMessage(), expectedExceptionMessage);
    }
  }

